lib/dateTime.js

```diff
@@ -0,0 +1,577 @@
+import {getDocumentLocaleSettings, getLanguage, merge} from './common.js';
+
+const hour24locales = ['da', 'de', 'es', 'fr', 'nl', 'sv', 'tr', 'zh'];
+const mondayFirstDayLocales = ['da', 'de', 'fr', 'nl', 'sv', 'tr'];
+const numericMonths = ['M01', 'M02', 'M03', 'M04', 'M05', 'M06', 'M07', 'M08', 'M09', 'M10', 'M11', 'M12'];
+
+function buildDayPeriodRe(part) {
+	let re = '';
+	let or = '';
+	for (let i = 0; i < part.length; i++) {
+		re += or + part.substr(0, i + 1);
+		or = '|';
+	}
+	return new RegExp(re, 'i');
+}
+
+function getParts() {
+
+	const descriptor = getDateTimeDescriptor();
+
+	const result = [];
+	const separator = getSeparator();
+	const parts = descriptor.formats.dateFormats.short.split(separator);
+
+	for (var i = 0; i < parts.length; i++) {
+		var part = parts[i].trim();
+		switch (part) {
+			case 'dd':
+			case 'd':
+				result.push('d');
+				break;
+			case 'MM':
+			case 'M':
+				result.push('M');
+				break;
+			case 'yyyy':
+				result.push('yyyy');
+				break;
+		}
+	}
+
+	if (result.length !== 3) {
+		return ['M', 'd', 'yyyy'];
+	}
+
+	return result;
+
+}
+
+const reSeparator = new RegExp('\\W');
+
+function getSeparator() {
+	const descriptor = getDateTimeDescriptor();
+	const match = reSeparator.exec(descriptor.formats.dateFormats.short);
+	if (match !== null) {
+		return match[0];
+	}
+	return '/';
+}
+
+function getTimeFormat(hour24, language, baseLanguage) {
+	let timeFormat = hour24 ? 'HH:mm' : 'h:mm tt';
+	if (baseLanguage === 'ja' || baseLanguage === 'pt' || baseLanguage === 'zh') {
```

Comment:
Does `es` also fit under this category?

Reply:
There also seems to be a difference between `pt` and `pt-BR`, unless I'm misunderstanding

Reply:
Right, so I'm actually not sure what to do here. If these changes should match what was previously in the library, you're correct.

The LMS has a DB table with all this stuff in it, but it only has entries for the specific locales -- `es-MX` and `pt-BR`. There's [this library](https://github.com/andyearnshaw/Intl.js/tree/master/locale-data/json) that's used as a polyfill and I think was what we originally used to populate these values, but the LMS isn't even consistent with it.

I think I'll fix this so that we're at least consistent with how things used to be, but in reality I'm not sure how these base languages should work (what regional locale "wins"?), or if we even really care since the LMS should never send down one of these language values anyway.

Reply:
Right yeah, I'm not sure what I should be comparing.  I'm also seeing a difference in https://github.com/Brightspace/intl/pull/31/files#diff-38bec192119c430c4cdc8729f798f951R317 and what used to be there, I don't know what we want to use as a source of truth

Reply:
Also a good catch -- that one was intentional as the data in here differed from the LMS for pretty much everything for the `en-GB` locale: full date, medium date, short date and month-day.

The other intentional changes were `pt-BR`'s month-year and month-day formats, and `sv-SE`'s month-day. Basically I re-aligned the stuff in here with what was in the LMS's database.

Reply:
Fixed the `es` and `pt` cases.

Reply:
Assuming French change was also on purpose (previously we had `fr` and `fr-CA` the same, but the database has them different)

Reply:
French was an interesting one, as we recently added `fr-FR` to the LMS database, which is the closest we're going to get to a base `fr` locale. So I just used its settings for the base `fr`, which yeah was a change since before we were treating `fr-CA` as the base one. Again, I don't think it really matters what the values are for these base ones... the only time they'll be used is if someone uses a custom locale that isn't a match (e.g. `fr-ZZ`).